solutions/usaco-763.mdx

```diff
@@ -2,147 +2,167 @@
+struct Log { int day, cow, change; };
 
-			change += L[i].c;
-			ids.put(L[i].id, change);
-			values.put(change, values.getOrDefault(change,0)+1);
+int main() {
+	std::ifstream read("measurement.in");
 
-			boolean isTop = change == values.firstKey();
-			int curCnt = values.get(change);
+	int n;
+	int g;
+	read >> n >> g;
 
-			if(wasTop){
-				if(isTop && curCnt == prevCnt && curCnt == 1){
-					continue;
-				}
-				//If it was the highest and now its not or there are now multiple highest cows, we need to increment answer
-				A++;
-			}
-			else if(isTop){
-				//If it wasn't at the highest but now it is, then we need to increment answer
-				A++;
-			}
-		}
-		PrintWriter out = new PrintWriter(new BufferedWriter(new FileWriter("measurement.out")));
-		out.println(A);
-		out.close();
+	vector<Log> log(n);
+	std::map<int, int> cows;
+	for (Log& l : log) {
+		read >> l.day >> l.cow >> l.change;
+		cows[l.cow] = g;
 	}
-	private static class Log{
-		int d, id, c;
-		public Log(int a, int b, int c){
-			d = a;
-			id = b;
-			this.c = c;
-		}
-	}
-	private static class InputReader {
-		public BufferedReader reader;
-		public StringTokenizer tokenizer;
-
-		public InputReader() {
-			try {
-				reader = new BufferedReader(new InputStreamReader(System.in), 32768);
-			} catch (Exception e) {
-				throw new NullPointerException("Could not create input stream");
-			}
+	std::sort(
+		log.begin(), log.end(),
+		[] (const Log& l1, const Log& l2) { return l1.day < l2.day; }
+	);
+
+	std::map<int, int> milk_prod{{g, n}};
+
+	int changeAmt = 0;
+	for (Log l : log) {
+		int milk_amt = cows[l.cow];
+		bool was_top = milk_amt == milk_prod.rbegin()->first;
+		int prev_count = milk_prod[milk_amt];
+		// remove the previous milk production number
+		milk_prod[milk_amt]--;
+		if (milk_prod[milk_amt] == 0) {
+			milk_prod.erase(milk_amt);
 		}
 
-		public InputReader(String fileName) {
-			try {
-				reader = new BufferedReader(new FileReader(new File(fileName)), 32768);
-			} catch (Exception ex) {
-				throw new NullPointerException("Input file does not exist! Put it in the project folder.");
-			}
-			tokenizer = null;
-		}
+		// update the milk production amounts
+		milk_amt += l.change;
+		cows[l.cow] = milk_amt;
+		milk_prod[milk_amt]++;
 
-		public String next() {
-			while (tokenizer == null || !tokenizer.hasMoreTokens()) {
-				try {
-					tokenizer = new StringTokenizer(reader.readLine());
-				} catch (IOException e) {
-					throw new RuntimeException(e);
-				}
+		bool is_top = milk_amt == milk_prod.rbegin()->first;
+		int curr_count = milk_prod[milk_amt];
+		if (was_top) {
+			if (is_top && curr_count == prev_count) {
+				continue;
 			}
-			return tokenizer.nextToken();
+			/*
+			 * if it was the highest and now it's not or there are now
+			 * multiple highest cows, then FJ needs to change the portrait
+			 */
+			changeAmt++;
+		} else if (is_top) {
+			/*
+			 * If it wasn't at the highest but now it is,
```

Comment:
would prefer casing to be consistent but it's eh

Reply:
will make a pr real quick